AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/MainActivity.kt

```diff
@@ -104,10 +105,15 @@ class MainActivity : AppCompatActivity(),
 
     override fun onItemClicked(item: ListItem)
     {
-        if (item is AdType)
+        if (item is AdType && item.intent != null)
         {
             startActivity(item.intent)
         }
+        else if (item is AdType && item.onTap != null)
```

Comment:
nit Since we have `item is AdType` in both conditions I usually abstract it away like:

```
if (item isAdType)
{
    if (item.intent != null )
    {
        startActivity(item.intent);
    }
    else if (item.runnable != null)
    {
        item.runnable();
    }
}
```

Reply:
ah ok, my previous places liked to keep it flat i.e. one layer of logic for if conditions